### PR TITLE
BE-1255: port azure integration

### PR DESCRIPTION
**Pull request by mdiazitexico**
**JIRA:** https://pinpt-hq.atlassian.net/browse/BE-1255

**Description:** add azure integration to agent

### CODE REVIEW

File: integrations/azure/azure.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+			Name:        "url",
+			Description: "URL to the Azure DevOps Server",
+			Required:    false,
+			Exportable:  true,
+			Default:     "https://dev.azure.com/",
+			Type:        reflect.TypeOf("string"),
+			Validator:   integration.NewURLValidator(),
+		},
+		&integration.ConfigKeyValue{
+			Name:        "vsurl",
```
Comment (josecordaz) [inline integrations/azure/azure.go]: In order to be consistentm we should name this `url` instead.

File: integrations/azure/export.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+var gitIntegration = git.Integration{
+	// Name of the integration
+	Name: "azure",
+	Init: func(logger log.Logger, url string, apikey string, storage *integration.Storage) (context.Context, error) {
+		_, ctx, err := api.GetAzureClient()
+		return ctx, err
+	},
+	FetchAllRepos: func() ([]*git.Repository, error) {
+		azureGitIntegration, ctx, _ := api.GetAzureClient()
+		return azureGitIntegration.FetchAllRepos(ctx) //TODO
```
Comment (josecordaz) [inline integrations/azure/export.go]: delete the `//TODO` text

File: integrations/azure/export.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+	},
+	// FetchRepo will retrieve the repo by name from Github
+	FetchRepo: func(repoName string) (*git.Repository, error) {
+		azureGitIntegration, ctx, _ := api.GetAzureClient()
+		return azureGitIntegration.FetchSingleRepo(ctx, repoName)
+	},
+	// FetchRepoCommits will retrieve all the commits for a branch in a repo
+	FetchRepoCommits: func(repoName string, projectV1 string, branchName string, object []byte, progress *progress.DependencyProgress) ([]*git.Commit, error) {
+		azureGitIntegration, ctx, _ := api.GetAzureClient()
+		return azureGitIntegration.FetchRepoCommits(ctx, repoName, branchName) //TODO
```
Comment (josecordaz) [inline integrations/azure/export.go]: same here

File: integrations/azure/azure.go (hunk 1 above)
Comment (josecordaz) [reply integrations/azure/azure.go]: i just so the `url` above, what is the `vsurl` about? both `url` and `vzurl` have the same description.

File: integrations/azure/export.go (hunk 2 above)
Comment (mdiazitexico) [reply integrations/azure/export.go]: Just removed these comments

File: integrations/azure/azure.go (hunk 1 above)
Comment (mdiazitexico) [reply integrations/azure/azure.go]: This is the Azure DevOps Services url; it is used in verification process to extract azure users via API (slightly different from the generic API url). Just updated the description to match this.

**Pull request merged**